odoo/tools/float_utils.py

```diff
@@ -71,47 +88,31 @@ def float_round(value, precision_digits=None, precision_rounding=None, rounding_
     # the order of magnitude of the value, to tip the tie-break in the right
     # direction.
     # Credit: discussion with OpenERP community members on bug 882036
-
-    normalized_value = value / rounding_factor # normalize
-    sign = math.copysign(1.0, normalized_value)
     epsilon_magnitude = math.log(abs(normalized_value), 2)
-    epsilon = 2**(epsilon_magnitude-52)
-
-    # TIE-BREAKING: UP/DOWN (for ceiling[resp. flooring] operations)
-    # When rounding the value up[resp. down], we instead subtract[resp. add] the epsilon value
-    # as the approximation of the real value may be slightly *above* the
-    # tie limit, this would result in incorrectly rounding up[resp. down] to the next number
-    # The math.ceil[resp. math.floor] operation is applied on the absolute value in order to
-    # round "away from zero" and not "towards infinity", then the sign is
-    # restored.
-
-    if rounding_method == 'UP':
-        normalized_value -= sign*epsilon
-        rounded_value = math.ceil(abs(normalized_value)) * sign
-
-    elif rounding_method == 'DOWN':
-        normalized_value += sign*epsilon
-        rounded_value = math.floor(abs(normalized_value)) * sign
-
-    # TIE-BREAKING: HALF-EVEN
-    # We want to apply HALF-EVEN tie-breaking rules, i.e. 0.5 rounds towards closest even number.
-    elif rounding_method == 'HALF-EVEN':
-        rounded_value = math.copysign(builtins.round(normalized_value), normalized_value)
-
-    # TIE-BREAKING: HALF-DOWN
-    # We want to apply HALF-DOWN tie-breaking rules, i.e. 0.5 rounds towards 0.
-    elif rounding_method == 'HALF-DOWN':
-        normalized_value -= math.copysign(epsilon, normalized_value)
-        rounded_value = round(normalized_value)
-
-    # TIE-BREAKING: HALF-UP (for normal rounding)
-    # We want to apply HALF-UP tie-breaking rules, i.e. 0.5 rounds away from 0.
-    else:
-        normalized_value += math.copysign(epsilon, normalized_value)
-        rounded_value = round(normalized_value)     # round to integer
-
-    result = rounded_value * rounding_factor # de-normalize
-    return result
+    # `epsilon_magnitude - 52` would be the minimal size, but we increase it to make it more
```

Comment:
```suggestion
    # `epsilon_magnitude - 50` would be the minimal size, but we increase it to make it more
```

Reply:
This comment is there to explain why it was changed from  `2**(epsilon_magnitude - 52)` (as suggested in the referenced [bug report](https://bugs.launchpad.net/openobject-server/+bug/882036)) to `2**(epsilon_magnitude - 50)`

Though I've just noticed that even [13 years ago](https://bugs.launchpad.net/openobject-server/+bug/882036/comments/28), it was noted `-52` was probably too small, and `-50` would still be safe to use.